SmartDeviceLink/SDLLifecycleManager.m

```diff
@@ -614,10 +631,23 @@ - (void)sendConnectionRequest:(__kindof SDLRPCRequest *)request withResponseHand
         if (handler) {
             handler(request, nil, [NSError sdl_lifecycle_notReadyError]);
         }
-
+        
         return;
     }
-
+    
+    if (!request.isPayloadProtected && [self.encryptionLifecycleManager rpcRequiresEncryption:request]) {
+        request.payloadProtected = YES;
+    }
+    
+    if (request.isPayloadProtected && !self.encryptionLifecycleManager.isEncryptionReady) {
+        SDLLogW(@"Encryption Manager not ready, request not sent (%@)", request);
+        if (handler) {
+            handler(request, nil, [NSError sdl_encryption_lifecycle_notReadyError]);
```

Comment:
I'd like to do option 1 but unfortunately it would be a breaking change to do the equivalent of completion handlers in Android. I think we should try to investigate option 2 where the operation queue is suspended until encryption is setup and see if it even works.

Reply:
Yes ive tested this. please see here: https://github.com/smartdevicelink/sdl_ios/pull/1320#discussion_r314551019